webpack/assets/javascripts/react_app/redux/reducers/notifications/index.js

```diff
@@ -25,12 +26,19 @@ export default (state = initialState, action) => {
     case NOTIFICATIONS_SET_EXPANDED_GROUP:
       return state.set('expandedGroup', payload.group);
     case NOTIFICATIONS_MARK_AS_READ:
+    return state.set(
+      'notifications',
+      state.notifications.map(
+        n => n.id === payload.id ?
+          Object.assign({}, n, {seen: true}) : n
+      )
+    );
+    case NOTIFICATIONS_MARK_GROUP_AS_READ:
       return state.set(
         'notifications',
         state.notifications.map(
-          n => n.id === payload.id ?
-            Object.assign({}, n, {seen: true}) :
-            n
+          n => payload.ids.includes(n.id) ?
```

Comment:
better to do `n => n.group === payload.group ?`
that way its more efficient - you dont have to pass around the ids and iterate on them